R/bayesfactor_parameters.R

```diff
@@ -209,22 +209,17 @@ bayesfactor_parameters.stanreg <- function(posterior,
                                            effects = c("fixed", "random", "all"),
                                            component = c("conditional", "zi", "zero_inflated", "all"),
                                            ...) {
+  cleaned_parameters <- insight::clean_parameters(posterior)
   effects <- match.arg(effects)
   component <- match.arg(component)
-  cleaned_parameters <- insight::clean_parameters(posterior)
-
-  # Get Priors
-  if (is.null(prior)) {
-    prior <- posterior
-  }
 
-  prior <- .update_to_priors(prior, verbose = verbose)
-  prior <- insight::get_parameters(prior, effects = effects, component = component)
-  posterior <- insight::get_parameters(posterior, effects = effects, component = component)
+  samps <- .clean_priors_and_posteriors(posterior, prior,
+                                        verbose = verbose,
+                                        effects = effects, component = component)
```

Comment:
Not sure what these changes all do, but I guess you know what you did? ;-)
If everything is working, I'd say these changes are ok, ready to merge.

Reply:
I hope I know what I'm doing ^_^

This and the following changes are just to reduce duplicated code the repeats in `bf_parametes` `bf_restrict` and `si`.